javascript/require-and-import-mini-lesson.md

```diff
@@ -0,0 +1,351 @@
+
+```bash
+node example-for-import.mjs
+```
+
+You should see:
+
+```bash
+This is a the first message!
+This is a the second message!
+```
+
+##### creating a `package.json` file
+
+Without delving to much into the details of what is a `package.json` file and why we use it, let's just simplify it. This file is kind of the blueprint of what your app will need to run properly. It can contain many things, depending on the app you will be creating. When you want to "activate" the blueprint, you simply run `npm install` and then you should see a file called `package-lock.json` appear, which will contain details on the installation you have just made.
+
+Now, don't worry if this seems a bit confusing, as we just need to use the file in order to set the environment and use `import` with a `.js` file instead of the `.mjs` that we created above.
+
+So, in order to do this, you create a subfolder (I called mine `import-with-package-json`) and copy the two files `my-module-for-import.mjs` and `example-for-import.mjs` and then rename them to have a `.js` extension. Depending on the code editor you are using, you might get a message asking you if you want to change the import statement. It's perfectly normal, just say no for now, we will change it manually.
+
+So, open the `example-for-imports.js` file and change the first line for this one:
+
+```javascript
+import { firstMessage, secondMessage} from './my-module-for-import.js';
+```
+
+Since we changed the file extension, we need to change it in our import.
+
+Next, we need to create the `package.json` file and put this code in:
+
+```javascript
+{
+  "type": "module"
+}
+```
+
+This simply will lay the grounds for our program to run as a module. To test this, inside of the new created folderwe first need to run the install command like I mentioned above:
+
+```bash
+npm install
+```
+
+NPM will run the installation and you can possibly see warnings or notices, don't mind them, it's fine for our purposes.
+
+When the installation is done, you can now run:
+
+```bash
+node example-for-import.js
+```
+
+And you should see:
+
+```bash
+This is a the first message!
+This is a the second message!
+```
+
+That's it! You now know how to use `import` with Node!
+
+Please note that the key thing with this is to always stay tuned to changes in the ECMA specifications, as they change often and the codebase progresses to bring in usually more depth and better performance. In my opinion, a good developer is a developer who's always keen to learn and is on the lookout for these changes!
+
+Now, pat yourself on the back, as you've gone through somewhat tough material. Go back, revise, try different things, and when you're ready, tackle on some of the challenges below!
+
+### Challenges
+
+- How could you change the line `const obj = require("./module-with-functions");` and change it to become more dynamic? Can you refactor it by using ES6 syntax? What's the benefit for this?
```

Comment:
What would a solution be to this? I'm not sure I understand what it's asking.

Reply:
removed